packages/fx-core/src/common/tools.ts

```diff
@@ -945,3 +947,32 @@ export async function getSPFxToken(
   }
   return spoToken;
 }
+
+export function getFixedCommonProjectSettings(rootPath: string | undefined) {
+  if (!rootPath) {
+    return undefined;
+  }
+
+  try {
+    const projectSettingsPath = path.join(
+      rootPath,
+      `.${ConfigFolderName}`,
+      InputConfigsFolderName,
+      ProjectSettingsFileName
+    );
+
+    if (!projectSettingsPath || !fs.pathExistsSync(projectSettingsPath)) {
+      return undefined;
+    }
+
+    const projectSettings = fs.readJsonSync(projectSettingsPath);
```

Comment:
Do we still need to do IO operation for read the project settings file every time? Thus we don't improve any performance?

Reply:
For fixed properties, it will be read only once